File: src/main/java/com/beemer/fromis9server/schedule/dto/ScheduleListDTO.java
package com.beemer.fromis9server.schedule.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ScheduleListDTO {
    private Long id;
    private LocalDateTime dateTime;
    private String schedule;
    private String description;
    @JsonProperty("icon")
    private ScheduleIconDTO scheduleIconDTO;
}

File: src/main/kotlin/com/beemer/unofficial/fromis9/schedule/dto/ScheduleListDto.kt
package com.beemer.unofficial.fromis9.schedule.dto

import java.time.LocalDateTime

data class ScheduleListDto(
    val page: SchedulePageDto,
    val schedules: List<ScheduleDetailsDto>
)

data class SchedulePageDto(
    val previousPage: Int?,
    val currentPage: Int,
    val nextPage: Int?,
)

data class ScheduleDetailsDto(
    val scheduleId: Int,
    val platform: String,
    val image: String,
    val color: String,
    val dateTime: LocalDateTime,
    val schedule: String,
    val description: String?,
    val url: String?,
    val allDay: Boolean
)